portfolio/src/main/webapp/script.js

```diff
@@ -17,7 +17,7 @@
 */
 
 function getMessage() {
-  fetch('/data').then(response => response.text()).then(msg => {
+  fetch('/data').then(response => response.json()).then(msg => {
```

Comment:
So now `msg` is a JSON array and you are implicitly depending on its `toString()` method to print it out.

You could make it better by constructing HTML elements for each of the message, but I won't block this PR for that.